src/test/java/uk/gov/register/presentation/functional/DataResourceTest.java

```diff
@@ -27,30 +29,24 @@ public static void publishTestMessages() {
     }
 
     @Test
-    public void downloadRegister_shouldReturnAZipfile() {
+    public void downloadRegister_shouldReturnAZipfile() throws IOException {
         Response response = getRequest("/download-register");
 
         assertThat(response.getHeaderString("Content-Type"), equalTo(MediaType.APPLICATION_OCTET_STREAM));
         assertThat(response.getHeaderString("Content-Disposition"), startsWith("attachment; filename="));
         assertThat(response.getHeaderString("Content-Disposition"), endsWith(".zip"));
         InputStream is = response.readEntity(InputStream.class);
-        ZipInputStream zis = new ZipInputStream(is);
-        ZipEntry entry;
-        boolean foundProofs = false;
-        boolean foundRegister = false;
-        try {
-            while ((entry = zis.getNextEntry()) != null) {
-                if (entry.getName().compareTo("proof.json") == 0) {
-                    foundProofs = true;
-                } else if (entry.getName().compareTo("register.json") == 0) {
-                    foundRegister = true;
-                }
-                System.out.println(entry.getName());
-            }
-        } catch (IOException e) {
-            fail("Unexpected IO exception - " + e.getMessage());
+        assertTrue(getEntries(is).contains("register.json"));
```

Comment:
can we stick to hamcrest please? Using hamcrest all the time results in consistent test failure reporting.  Also, when there's a failure you get something more than just "wanted true, got false" -- it will report what the list actually contained and not just what was being looked for.

```json
assertThat(getEntries(is), hasItem("register.json"))
```